### PR TITLE
Replace SA, SB and SE cms calls with the latest  api endpoints

### PR DESCRIPTION
**Pull request by pkfec**
## Summary (required)
Replace SA, SB and SE api calls with the latest api endpoints. A total of  7 endpoints are updated

- Resolves https://github.com/fecgov/fec-cms/issues/3285

## Impacted areas of the application
Committee Raising (SA), Spending (SB) and Independent Expenditure (SE) api calls (endpoints).

## Screenshots
**Committee raising tab screen:**

Screen Shot 2019-11-08 at 3 40 00 PM
#########################################################################

** Raising tab api calls:** ::: #4 endpoints are updated
#########################################################################
Before:
https://fec-dev-api.app.cloud.gov/v1/committee/C00448696/schedules/schedule_a/by_state/?api_key=pjE98TJBLgSA1XyqNHVMgt1TK88S7qtpIklwztqW&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=99&page=1&cycle=2020&election_full=false

After:
https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_a/by_state/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=99&page=1&cycle=2020&election_full=false&committee_id=C00448696
#########################################################################
Before:
https://fec-dev-api.app.cloud.gov/v1/committee/C00448696/schedules/schedule_a/by_size/?api_key=pjE98TJBLgSA1XyqNHVMgt1TK88S7qtpIklwztqW&sort_hide_null=false&sort_nulls_last=true&per_page=10&page=1&cycle=2020&election_full=false

After:
https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_a/by_size/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=true&per_page=10&page=1&committee_id=C00448696&cycle=2020
#########################################################################
Before:
https://fec-dev-api.app.cloud.gov/v1/committee/C00448696/schedules/schedule_a/by_employer/?api_key=pjE98TJBLgSA1XyqNHVMgt1TK88S7qtpIklwztqW&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=10&page=1&cycle=2020&election_full=false

After:
https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_a/by_employer/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=10&page=1&committee_id=C00448696&cycle=2020
#########################################################################
Before:
https://fec-dev-api.app.cloud.gov/v1/committee/C00448696/schedules/schedule_a/by_occupation/?api_key=pjE98TJBLgSA1XyqNHVMgt1TK88S7qtpIklwztqW&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=10&page=1&cycle=2020&election_full=false

After:
https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_a/by_occupation/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=10&page=1&committee_id=C00448696&cycle=2020

#########################################################################

**Spending tab screen**:
Screen Shot 2019-11-08 at 4 01 12 PM

**Spending tab api calls:** ::: #3 endpoints are updated
########################################################################
Before:
https://fec-dev-api.app.cloud.gov/v1/committee/C00448696/schedules/schedule_e/by_candidate/?api_key=pjE98TJBLgSA1XyqNHVMgt1TK88S7qtpIklwztqW&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=30&page=1&cycle=2020&election_full=false

After:
https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_e/by_candidate/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=30&page=1&committee_id=C00448696&cycle=2020&election_full=false
#########################################################################
Before:
https://fec-dev-api.app.cloud.gov/v1/committee/C00448696/schedules/schedule_b/by_recipient/?api_key=pjE98TJBLgSA1XyqNHVMgt1TK88S7qtpIklwztqW&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=10&page=1&cycle=2020&election_full=false

After:
https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_b/by_recipient/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=10&page=1&committee_id=C00448696&cycle=2020

#########################################################################
Before:
https://fec-dev-api.app.cloud.gov/v1/committee/C00448696/schedules/schedule_b/by_recipient_id/?api_key=pjE98TJBLgSA1XyqNHVMgt1TK88S7qtpIklwztqW&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=10&page=1&cycle=2020&election_full=false

After:
https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_b/by_recipient_id/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=10&page=1&committee_id=C00448696&cycle=2020

#########################################################################

## How to test
- checkout the branch
- export FEC_API_URL to `dev` or `stage` api
- run server
-  search for a committee and look at the raising, spending tabs results should load. Observe the api calls in the console.

### CODE REVIEW

**Comment by fec-jli** [comment]
When clicking Raising, the last url still point to  committee/committee_id/schedules/schedule_a/by_state. It looks like useless call? not sure.

by_size

**Comment by pkfec** [comment]
> When clicking Raising, the last url (mouse over on it) still points to committee/committee_id/schedules/schedule_a/by_state. It looks like useless call? not sure.
> Could you look at it?
> 
> by_size

Good catch @fec-jli. I replaced the deprecated endpoint with the latest api call. Please test again.

**Comment by pkfec** [comment]
> Requesting a JavaScript tweak but in a few different places.
> 
> Additionally, I don't know if it matters, but the first receipts call in the PR isn't sending what you have listed.
> 
> `https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_a/by_state/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=99&page=1&cycle=2020&election_full=false&committee_id=C00448696`
> 
> is only sending `committee_id`, `cycle`, and `per_page`.
> 
> `api_key` is going as `''` but that's fine

@rfultz  there are two api `schedules/schedule_a/by_state/` calls on the
 Raising tab/ Individual contributions/ Group by State:

Call 1: This displays the amount per state in for given committee in a table
https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_a/by_state/?api_key=DEMO_KEY&sort_hide_null=false&sort_nulls_last=true&sort=-total&per_page=99&page=1&cycle=2020&election_full=false&committee_id=C00448696

Call 2:  Display  amounts per `state` on the map. I do see an `api_key` being set on the request.Could you please check this again?
https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_a/by_state/?api_key=DEMO_KEY&committee_id=C00448696&cycle=2020&per_page=99

File: fec/fec/static/js/pages/committee-single.js [hunk 1]
```diff
@@ -398,10 +398,13 @@ $(document).ready(function() {
     };
     switch ($table.attr('data-type')) {
       case 'contribution-size':
-        path = ['committee', committeeId, 'schedules', 'schedule_a', 'by_size'];
+        path = ['schedules', 'schedule_a', 'by_size'];
         tables.DataTable.defer($table, {
           path: path,
-          query: query,
+          query: {
```
[the remark this answers is not in the document]
Comment (pkfec) [reply fec/fec/static/js/pages/committee-single.js]: Good point @rfultz  I will make the changes.

File: fec/fec/static/js/pages/committee-single.js [hunk 2]
```diff
@@ -472,18 +468,15 @@ $(document).ready(function() {
...
-          'schedule_a',
-          'by_occupation'
-        ];
+        path = ['schedules', 'schedule_a', 'by_occupation'];
         tables.DataTable.defer(
           $table,
           _.extend({}, tableOpts, {
             path: path,
-            query: query,
+            query: {
```
[the remark this answers is not in the document]
Comment (pkfec) [reply fec/fec/static/js/pages/committee-single.js]: will do

File: fec/fec/static/js/pages/committee-single.js [hunk 3]
```diff
@@ -523,18 +516,15 @@ $(document).ready(function() {
...
-          'schedule_b',
-          'by_recipient'
-        ];
+        path = ['schedules', 'schedule_b', 'by_recipient'];
         tables.DataTable.defer(
           $table,
           _.extend({}, tableOpts, {
             path: path,
-            query: query,
+            query: {
```
[the remark this answers is not in the document]
Comment (pkfec) [reply fec/fec/static/js/pages/committee-single.js]: will do

File: fec/fec/static/js/pages/committee-single.js [hunk 4]
```diff
@@ -598,18 +588,15 @@ $(document).ready(function() {
...
-          'schedule_b',
-          'by_recipient_id'
-        ];
+        path = ['schedules', 'schedule_b', 'by_recipient_id'];
         tables.DataTable.defer(
           $table,
           _.extend({}, tableOpts, {
             path: path,
-            query: query,
+            query: {
```
[the remark this answers is not in the document]
Comment (pkfec) [reply fec/fec/static/js/pages/committee-single.js]: yes, will do

File: fec/fec/static/js/pages/committee-single.js [hunk 5]
```diff
@@ -623,16 +610,14 @@ $(document).ready(function() {
...
-          committeeId,
-          'schedules',
-          'schedule_e',
-          'by_candidate'
-        ];
+        path = ['schedules', 'schedule_e', 'by_candidate'];
         tables.DataTable.defer($table, {
           path: path,
-          query: query,
+          query: {
```
[the remark this answers is not in the document]
Comment (pkfec) [reply fec/fec/static/js/pages/committee-single.js]: yes, i will update the code.

**Pull request merged**